### DISCUSSION TITLE
cmake include_directores path can't be resolved by clangd

### DISCUSSION THREAD

**Discussion by CalebDu** [clangd/clangd#1649] [Q&A]
I include several 3party header-only library by include_directores() in cmake. but clangd can't find these paths correctly in vscode.  after i add  compiler flag "-I <paths>" in compiler_commands.json manully. it works.
So how to handle this problem automaticly?

**Comment by i-ky** [comment]
You should use CMake to generate `compile_commands.json`. If you do so already and it does not work without manual intervention, then it may be a CMake bug.

**Reply by CalebDu** [reply]
cmake has generate comiple_commands.json without include_directories path. so i'm confused that how to make  cmake add -I <path> in comiple_commands.json automaticly.  otherwise, clangd can't resolve these 3party libraries.  Maybe i should post this disscussion in cmake repo?

**Reply by i-ky** [reply]
Just to make sure... Have you re-run CMake after you added `include_directories()`?

> Maybe i should post this disscussion in cmake repo?

This may be a good idea. Your question is more related to CMake than clangd. I think you will be able to get help faster among CMake users.

**Reply by CalebDu** [reply]
I downgraded cmake and the problem disappeared.
